src/common/get-file-info.js

```diff
@@ -0,0 +1,38 @@
+"use strict";
+
+const createIgnorer = require("../common/create-ignorer");
+const fs = require("fs");
+const options = require("../main/options");
+
+/** @param {{filepath: string, ignorePath: string | false = '.prettierignore', withNodeModules: boolean = false, ...otherPrettierOptions }} opts */
+function getFileInfo(opts) {
+  let stats;
+  try {
+    stats = fs.statSync(opts.filepath);
+  } catch (e) {
+    // file / dir at filePath does not exist - no need to handle
+  }
+  const exists = (stats && stats.isFile()) || false;
+
+  const ignorePath =
+    typeof opts.ignorePath !== "undefined"
+      ? opts.ignorePath
+      : ".prettierignore";
+
+  let ignored = false;
+  if (ignorePath) {
+    const ignorer = createIgnorer(ignorePath, opts.withNodeModules);
+    ignored = ignorer.ignores(opts.filepath);
+  }
+
+  const inferredParser =
+    options.inferParser(opts.filepath, opts.plugins) || null;
+
+  return {
+    exists,
+    ignored,
+    inferredParser
+  };
+}
```

Comment:
My opinions:

- `getFileInfo(filepath, {ignorePath, withNodeModules, plugins})`
- returns `{ignored, inferredParser}`

Reply:
> `getFileInfo(filepath, {ignorePath, withNodeModules, plugins})`

Looks a bit simpler to use and does not expose different naming styles for `filepath` and `ignorePath` – I like this! Will change the API again as you suggest. We can also avoid setting `ignorePath` when a method is called directly (not via CLI) – that may work a bit more predictably for Prettier API users. The only problem that remains is with plugins – they will not be found automatically unless `cwd` matches project path (which will be the case for editor plugins). We need https://github.com/prettier/prettier/pull/4192 to solve that.

> Last suggestion: isFormattable that returns true if path is not ignored and a parser can be inferred
see https://github.com/prettier/prettier/pull/4341#issuecomment-383473937 and https://github.com/prettier/prettier/pull/4341#issuecomment-382638660

Reply:
Sorry I wasn't clear. I'm suggesting an alternative API to `getFileInfo` not an additional return value.

What I mean is: most integrations will use this function and use something like:
```js
if (!fileinfo.ignored && fileinfo.parser) { const formatted = prettier.format(text, {options}); }
```

So maybe we could do instead:

```js
if (!prettier.isFileSupported(filepath, someOptions)) return;
const formatted = prettier.format(text, {options});
```

WDYT?

Reply:
What you’re suggesting when work when using format on save. But when a person calls format manually, I can imagine a scenario when .prettierifnore is ignored. I’d let editors decide what to do in this case.

Reply:
Hmm in the second scenario we can just avoid passing .prettierignore as an option and this will do the job. Guess you’re right and I should open a competing PR with your suggestion.

Reply:
I don't know if we need separate PRs, they'll probably be very similar, we just need to figure out the API but it's your call!

> What would be an appropriate CLI option name in this case?

Now you got me :joy: